blueprints/src/main/java/dev/streamx/aem/connector/blueprints/ReferencedAssetHandler.java

```diff
@@ -0,0 +1,98 @@
+package dev.streamx.aem.connector.blueprints;
+
+import dev.streamx.blueprints.data.Asset;
+import dev.streamx.sling.connector.PublicationHandler;
+import dev.streamx.sling.connector.handling.Configuration;
+import dev.streamx.sling.connector.handling.InternalResourcesHandler;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.apache.sling.engine.SlingRequestProcessor;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.metatype.annotations.Designate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(
+    service = {
+        InternalResourcesHandler.class, PublicationHandler.class, ReferencedAssetHandler.class
+    },
+    immediate = true,
+    configurationPolicy = ConfigurationPolicy.REQUIRE
+)
+@Designate(ocd = ReferencedAssetHandlerConfig.class)
+public class ReferencedAssetHandler extends InternalResourcesHandler<Asset> {
```

Comment:
I guess we don't need to assume in the name that source of this publish is reference.  Could be AssetResourcePathPublicationHandler (according to my proposal of InternalResourcesHandler -> ResourcePathPublicationHandler)